includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -68,24 +112,32 @@ public static function filter_event_body( $body, $event_name ) {
 	/**
 	 * Handler for the reader_logged_in event.
 	 *
-	 * @param int   $timestamp Timestamp of the event.
-	 * @param array $data      Data associated with the event.
-	 * @param int   $user_id   ID of the client that triggered the event. It's a RAS ID.
+	 * @param int   $params The GA4 event parameters.
+	 * @param array $data      Data associated with the Data Events api event.
 	 *
-	 * @throws \Exception If the event is invalid.
-	 * @return void
+	 * @return array $params The final version of the GA4 event params that will be sent to GA.
 	 */
-	public static function handle_reader_logged_in( $timestamp, $data, $user_id ) {
-
-		$params    = $data['ga_params'];
-		$client_id = $data['ga_client_id'];
+	public static function handle_reader_logged_in( $params, $data ) {
+		return $params;
+	}
 
-		if ( empty( $client_id ) ) {
-			throw new \Exception( 'Missing client ID' );
+	/**
+	 * Handler for the reader_registered event.
+	 *
+	 * @param int   $params The GA4 event parameters.
+	 * @param array $data      Data associated with the Data Events api event.
+	 *
+	 * @return array $params The final version of the GA4 event params that will be sent to GA.
+	 */
+	public static function handle_reader_registered( $params, $data ) {
+		$params['registration_method'] = $data['metadata']['registration_method'] ?? '';
+		if ( ! empty( $data['metadata']['newspack_popup_id'] ) ) {
+			$params = array_merge( $params, Popups_Events::get_popup_metadata( $data['metadata']['newspack_popup_id'] ) );
 		}
```

Comment:
Don't know the reason, but the `newspack_popup_id` is not being populated to my registration through a campaigns popup. This is the resulting params from a `center overlay` campaign:

```
(
    [logged_in] => yes
    [is_reader] => yes
    [registration_method] => registration-block
    [referer] => /
)
```

Reply:
Check if you have blocks and popups plugins in the latest master and that they are built